quic/s2n-quic-rustls/src/certificate.rs

```diff
@@ -81,6 +81,19 @@ cert_type!(
     Vec<rustls::Certificate>
 );
 
+impl IntoCertificate for Vec<Vec<u8>> {
+    fn into_certificate(self) -> Result<Certificate, Error> {
+        let certs = self
+            .iter()
+            .map(|cert| der::into_certificate(cert.to_vec()))
+            .collect::<Result<Vec<_>, _>>()?
+            .into_iter()
+            .flatten()
+            .collect();
```

Comment:
I'm wondering if this code would be clearer with a more imperative approach?
```suggestion
        let mut certs = vec![];
        for der in self {
            let der = der.into_certificate()?;
            certs.extend(der.0);
        }
```

Reply:
yeah, much clearer!